File: mobile/app/src/main/java/com/auratherm/mobile/model/Sensor.kt
package com.auratherm.mobile.model

import com.google.gson.annotations.SerializedName

data class Sensor(
    val id: String,
    val name: String,
    val location: String?,
    val status: String,

    @SerializedName("zone_id")
    val zoneId: String,

    @SerializedName("store_id")
    val storeId: String,

    val zone: Zone, // Вкладена інформація про зону

    // Бекенд завжди повертає список (навіть якщо там 1 елемент або порожньо)
    val readings: List<SensorReading>
) {
    // Зручна властивість для UI: одразу повертає останню температуру або null, якщо вимірювань ще не було
    val currentTemperature: Double?
        get() = readings.firstOrNull()?.temperature

    val currentHumidity: Double?
        get() = readings.firstOrNull()?.humidity
}

File: frontend/src/types/sensor.ts
export type SensorStatus = "ACTIVE" | "INACTIVE" | "MAINTENANCE";

export interface SensorReading {
  id?: string;
  temperature: number;
  humidity?: number | null;
  dew_point?: number | null;
  timestamp: string;
  sensor_id?: string;
}

export interface Sensor {
  id: string;
  name: string;
  location?: string | null;
  pos_x?: number | null;
  pos_y?: number | null;
  status: SensorStatus;
  zone_id: string;
  store_id: string;
  // Зона з лімітами
  zone?: {
    id: string;
    name: string;
    min_temp: number;
    max_temp: number;
    store_id: string;
  };
  // Працівники, прикріплені до датчика
  assignedTo?: {
    user: {
      id: string;
      first_name: string;
      last_name: string;
    };
  }[];
  // Масив з останнім показником (або історією)
  readings?: SensorReading[];
}
